samples/csharp_dotnetcore/11.qnamaker/QnAMaker/BotServices.cs

```diff
@@ -19,10 +20,51 @@ public class BotServices
         /// <summary>
         /// Initializes a new instance of the <see cref="BotServices"/> class.
         /// </summary>
-        /// <param name="qnaServices">A dictionary of named <see cref="QnAMaker"/> instances for usage within the bot.</param>
-        public BotServices(Dictionary<string, QnAMaker> qnaServices)
+        /// <param name="botConfiguration">A dictionary of named <see cref="BotConfiguration"/> instances for usage within the bot.</param>
+        public BotServices(BotConfiguration botConfiguration)
         {
-            QnAServices = qnaServices ?? throw new ArgumentNullException(nameof(qnaServices));
+            foreach (var service in botConfiguration.Services)
+            {
+                switch (service.Type)
+                {
+                    case ServiceTypes.QnA:
+                        {
+                            // Create a QnA Maker that is initialized and suitable for passing
+                            // into the IBot-derived class (QnABot).
+                            var qna = (QnAMakerService)service;
```

Comment:
This should be:
`var qna = service as QnaMakerService; `
The cast as you have it now will throw, rather than return null.

Reply:
(I see that's from the old code, but... a bug is a bug is a bug... ) :)

Reply:
Indeed, fixing it.